### PR TITLE
Add Validations and model specs

### PR DESCRIPTION
**Pull request by JonahKayizzi**
This pull request consists of the following;

- Added the following validations:

- [ ] For the User model:

`Name must not be blank.`
`PostsCounter must be an integer greater than or equal to zero.`

- [ ] For the Post model:

`Title must not be blank.`
`Title must not exceed 250 characters.`
`CommentsCounter must be an integer greater than or equal to zero.`
`LikesCounter must be an integer greater than or equal to zero.`

- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

**Review by codecaiine — COMMENTED** [review]
Hi @JonahKayizzi , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that the readme file is not complete. Please make sure that your readme has all the main sections like **future features section** and add 1 - 3 future features keys. And also please fix the link of your license and your logo which are broken. Also please make sure that the logo you are using is your own logo. If it is microverse logo please remove it as required  :+1: 


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  end
+
+  it 'Update posts counter for user' do
+    expect(subject.user.PostsCounter).to eq(1)
+  end
+
+  it 'Return five most recent comments for post' do
+    expect(subject.five_most_recent_comments_for_post).to eq(subject.comments.order(created_at: :desc).limit(5))
+  end
+end
```
Comment (DeliceLydia) [inline spec/post_spec.rb]: - [ ] Some tests are not passing please fix them
Screen Shot 2023-01-21 at 17 10 26

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by rotimiazeez — APPROVED** [review]
Hi @JonahKayizzi  👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Tests are passing ✔️ 
- Descriptive README 😉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**